src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Bjorhus.cpp

```diff
@@ -0,0 +1,488 @@
+        projection_ab,
+    const gsl::not_null<tnsr::Ab<DataVector, Dim, Frame::Inertial>*>
+        projection_Ab,
+    const gsl::not_null<tnsr::AA<DataVector, Dim, Frame::Inertial>*>
+        projection_AB,
+    const gsl::not_null<tnsr::aa<DataVector, Dim, Frame::Inertial>*>
+        char_projected_rhs_dt_v_psi,
+    const gsl::not_null<tnsr::iaa<DataVector, Dim, Frame::Inertial>*>
+        char_projected_rhs_dt_v_zero,
+    const gsl::not_null<tnsr::aa<DataVector, Dim, Frame::Inertial>*>
+        char_projected_rhs_dt_v_minus,
+    const gsl::not_null<tnsr::a<DataVector, Dim, Frame::Inertial>*>
+        constraint_char_zero_plus,
+    const gsl::not_null<tnsr::a<DataVector, Dim, Frame::Inertial>*>
+        constraint_char_zero_minus,
+    const gsl::not_null<std::array<DataVector, 4>*> char_speeds,
+
+    const std::optional<tnsr::I<DataVector, Dim, Frame::Inertial>>&
+    /* face_mesh_velocity */,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& normal_covector,
+    const tnsr::aa<DataVector, Dim, Frame::Inertial>& pi,
+    const tnsr::iaa<DataVector, Dim, Frame::Inertial>& phi,
+    const tnsr::aa<DataVector, Dim, Frame::Inertial>& spacetime_metric,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& /* coords */,
+    const Scalar<DataVector>& gamma1, const Scalar<DataVector>& gamma2,
+    const Scalar<DataVector>& lapse,
+    const tnsr::a<DataVector, Dim, Frame::Inertial>& gauge_source,
+    const tnsr::ab<DataVector, Dim, Frame::Inertial>&
+        spacetime_deriv_gauge_source,
+    const tnsr::aa<DataVector, Dim, Frame::Inertial>& dt_pi,
+    const tnsr::iaa<DataVector, Dim, Frame::Inertial>& dt_phi,
+    const tnsr::aa<DataVector, Dim, Frame::Inertial>& dt_spacetime_metric,
+    const tnsr::iaa<DataVector, Dim, Frame::Inertial>& d_pi,
+    const tnsr::ijaa<DataVector, Dim, Frame::Inertial>& d_phi,
+    const tnsr::iaa<DataVector, Dim, Frame::Inertial>& d_spacetime_metric)
+    const noexcept {
+  TempBuffer<tmpl::list<::Tags::TempScalar<0, DataVector>,
+                        ::Tags::TempI<0, Dim, Frame::Inertial, DataVector>,
+                        ::Tags::Tempii<0, Dim, Frame::Inertial, DataVector>,
+                        ::Tags::Tempa<0, Dim, Frame::Inertial, DataVector>,
+                        ::Tags::Tempia<0, Dim, Frame::Inertial, DataVector>>>
+      local_buffer(get_size(get<0>(normal_covector)));
+
+  //   auto& lapse = get<::Tags::TempScalar<0, DataVector>>(local_buffer);
+  auto& shift =
+      get<::Tags::TempI<0, Dim, Frame::Inertial, DataVector>>(local_buffer);
+  auto& spatial_metric =
+      get<::Tags::Tempii<0, Dim, Frame::Inertial, DataVector>>(local_buffer);
+  auto& spacetime_unit_normal_one_form =
+      get<::Tags::Tempa<0, Dim, Frame::Inertial, DataVector>>(local_buffer);
+  auto& two_index_constraint =
+      get<::Tags::Tempia<0, Dim, Frame::Inertial, DataVector>>(local_buffer);
+
+  *inverse_spacetime_metric = determinant_and_inverse(spacetime_metric).second;
```

Comment:
I think the advice is usually to use the inverse spatial metric to compute the inverse spacetime metric rather than getting it from direct inversion

Reply:
makes sense, changed it to avoid inversion